collect_app/src/test/java/org/odk/collect/android/preferences/dialogs/ResetProgressDialogTest.kt

```diff
@@ -0,0 +1,50 @@
+package org.odk.collect.android.preferences.dialogs
+
+import android.content.Context
+import android.view.View
+import android.widget.TextView
+import androidx.appcompat.app.AlertDialog
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.odk.collect.fragmentstest.DialogFragmentTest
+import org.odk.collect.permissions.R
+import org.odk.collect.strings.localization.getLocalizedString
+
+@RunWith(AndroidJUnit4::class)
+class ResetProgressDialogTest {
+    private val context = ApplicationProvider.getApplicationContext<Context>()
+
+    @Test
+    fun `The dialog should not be dismissed after clicking out of its area or on device back button`() {
+        val scenario = DialogFragmentTest.launchDialogFragment(ResetProgressDialog::class.java)
+        scenario.onFragment {
+            assertThat(it.isCancelable, `is`(false))
+        }
+    }
+
+    @Test
+    fun `The dialog should display proper content`() {
+        val scenario = DialogFragmentTest.launchDialogFragment(ResetProgressDialog::class.java)
+        scenario.onFragment {
+            // Button positive
+            assertThat((it.dialog!! as AlertDialog).getButton((AlertDialog.BUTTON_POSITIVE)).visibility, `is`(View.GONE))
+
+            // Button neutral
+            assertThat((it.dialog!! as AlertDialog).getButton((AlertDialog.BUTTON_NEUTRAL)).visibility, `is`(View.GONE))
+
+            // Button negative
+            assertThat((it.dialog!! as AlertDialog).getButton((AlertDialog.BUTTON_NEGATIVE)).visibility, `is`(View.GONE))
+
+            // Title
+            val titleId: Int = context.resources.getIdentifier("alertTitle", "id", context.packageName)
```

Comment:
We could make it much easier to test `ProgressDialogFragment` implementations by simply adding getters for title and message - these being rendered properly is tested in `ProgressDialogFragmentTest`. Let's add them now. I think given that we're testing that this is a correctly configured `ProgressDialogFragment` we also don't need to test the buttons (as they wouldn't be there by default).

Reply:
You mean adding getters for `title` and `message` to `ProgressDialogFragmentTest` so that we can use it in all implementations of `ProgressDialogFragmentTest`?